File: upvpn-server/src/conversions/location.rs
impl From<crate::proto::Location> for upvpn_types::location::Location {
    fn from(value: crate::proto::Location) -> Self {
        Self {
            code: value.code,
            country: value.country,
            country_code: value.country_code,
            city: value.city,
            city_code: value.city_code,
            state: value.state,
            state_code: value.state_code,
        }
    }
}

impl From<crate::proto::ListLocationsResponse> for Vec<upvpn_types::location::Location> {
    fn from(value: crate::proto::ListLocationsResponse) -> Self {
        value
            .locations
            .into_iter()
            .map(upvpn_types::location::Location::from)
            .collect()
    }
}

File: upvpn-android/app/src/main/kotlin/app/upvpn/upvpn/data/db/Location.kt
package app.upvpn.upvpn.data.db

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "location")
data class Location(
    @PrimaryKey(autoGenerate = false)
    val code: String,
    val country: String,
    val countryCode: String,
    val city: String,
    val cityCode: String,
    val state: String? = null,
    val stateCode: String? = null,
    val lastAccess: Int? = null
)
